### PR TITLE
Adds Remote campus support

### PR DESCRIPTION
**Pull request by vinnyjth**
### What does this PR do, or why is it needed?

On at least two of our other apps (LCBC, The Porch), we have code that supports remote campuses. The code used is identical in both cases, and doesn't have any negative trade offs, so I figured it would be handy to bring it over to core.

### What design trade-offs/decisions were made?

n/a

### How do I test this PR?

Download LCBC or The Porch ;) My tests should also show the code functioning. 

### What automated tests did you write?

Updated existing tests and added new tests.

## TODO:

- [x] PR has a relevant title that will be understandable in a public changelog (ie...non developers)
- [x] No new warnings in tests, in storybook, and in-app
- [ ] Upload GIF(s) of iOS and Android
- [x] Set a relevant reviewer

## REVIEW:

**Manual QA**

- [ ] Manual QA on iOS and ensure it looks/behaves as expected
- [ ] Manual QA on Android and ensure it looks/behaves as expected

**Code Review: Questions to consider**

- [ ] Read through the "Files changed" tab _very carefully_
- [ ] Edge cases: what assumptions are made about input?
- [ ] What kind of tests could be written?
- [ ] How might we make this easier for someone else to understand?
- [ ] Could the code be simpler?
- [ ] Will the code be easy to modify in the future?
- [ ] What's one part of these changes that makes you excited to merge it?

> The purpose of PR Review is to _improve the quality of the software._

### CODE REVIEW

File: packages/apollos-apollo-server-env-mock/src/index.js [hunk 1]
```diff
@@ -256,9 +256,9 @@ const createApolloServerEnvMock = (apolloServerEnv) => {
         return resolveWith([]);
       }
 
-      if (url.match('api/Campuses/\\d')) {
-        return resolveWith(rockMocks.campuses()[0]);
-      }
+      // if (url.match('api/Campuses/\\d')) {
```
Comment (conrad-vanl) [inline packages/apollos-apollo-server-env-mock/src/index.js:259]: Can we remove this code altogether?

File: packages/apollos-data-connector-rock/src/campuses/data-source.js [hunk 2]
```diff
@@ -9,21 +9,40 @@ export default class Campus extends RockApolloDataSource {
 
   getFromId = (id) =>
     this.request()
-      .find(id)
+      .filter(`Id eq ${id}`)
```
Comment (conrad-vanl) [inline packages/apollos-data-connector-rock/src/campuses/data-source.js:12]: Is there a reason we move from `find` to `.filter(...).first()`? Does `expand('CampusTypeValue)` not work on `find()` I'm guessing?

File: packages/apollos-data-connector-rock/src/campuses/data-source.js [hunk 3]
```diff
@@ -9,21 +9,40 @@ export default class Campus extends RockApolloDataSource {
...
   getByLocation = async ({ latitude, longitude } = {}) => {
     let campuses = await this.getAll();
+
+    const onlineCampuses = campuses
+      .filter(({ campusTypeValue }) => campusTypeValue?.value === 'Online')
+      .map((campus) => ({
+        ...campus,
+        location: {
+          ...campus.location,
+          street1: 'No locations near you. ',
```
Comment (conrad-vanl) [inline packages/apollos-data-connector-rock/src/campuses/data-source.js:36]: These values should really come from the config.yaml....

File: packages/apollos-data-connector-rock/src/campuses/data-source.js [hunk 4]
```diff
@@ -9,21 +9,40 @@ export default class Campus extends RockApolloDataSource {
...
+          city: "When there's one",
+          state: "we'll let you know!",
```
Comment (conrad-vanl) [inline packages/apollos-data-connector-rock/src/campuses/data-source.js:38]: Also, I kind of wonder if we should generalize this field or something.... deprecate `location` and make like `locationText` that allows for line breaks and such?

File: packages/apollos-apollo-server-env-mock/src/index.js (hunk 1 above)
Comment (vinnyjth) [reply packages/apollos-apollo-server-env-mock/src/index.js:259]: Yep

File: packages/apollos-data-connector-rock/src/campuses/data-source.js (hunk 2 above)
Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/campuses/data-source.js:12]: Yah, `expand` doesn't work at all with `.find`. None of the extra data was never being fetched.

File: packages/apollos-data-connector-rock/src/campuses/data-source.js (hunk 4 above)
Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/campuses/data-source.js:38]: I'm on board with that. I don't think we actually use any of the fields for geolocation besides lat/long

File: packages/apollos-data-connector-rock/src/campuses/data-source.js (hunk 3 above)
Comment (vinnyjth) [reply packages/apollos-data-connector-rock/src/campuses/data-source.js]: What do you feel about something like this? https://github.com/ApollosProject/apollos-apps/pull/1520/commits/7119dc0439635d49200ed5cff8257ae1a91fa14b

Adds ability to modify those fields easily, as needed, and without needing to change all of them. Also is backwards compat because we aren't required to add new mandatory fields to the config.yml
